controllers/pvc-mutator/storageclass/annotation_test.go

```diff
@@ -0,0 +1,212 @@
+package storageclass
+
+import (
+	"context"
+	"testing"
+
+	"github.com/storageos/api-manager/internal/pkg/provisioner"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestMutatePVCErrorToFetchStorageClass(t *testing.T) {
+	// Create a new scheme and add all the types from different clientsets.
+	scheme := runtime.NewScheme()
+	if err := kscheme.AddToScheme(scheme); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a fake client to fail on get StorageClass.
+	k8s := fake.NewClientBuilder().Build()
+
+	// Create a AnnotationSetter instance with the fake client.
+	annotationSetter := AnnotationSetter{
+		Client: k8s,
+		log:    ctrl.Log,
+	}
+
+	namespace := "namespace"
+
+	// Create a pvc.
+	pvc := createPVC("pvc1", namespace, nil)
+
+	err := annotationSetter.MutatePVC(context.Background(), pvc, namespace)
+	if err == nil {
+		t.Fatal("this must fail")
```

Comment:
The message doesn't make it clear why the test failed. Like in the other tests, it'd be better to have some proper contextual info, as simple as "expected error, got none".